util/helm/cmd.go

```diff
@@ -342,17 +342,28 @@ type TemplateOpts struct {
 	SkipCrds    bool
 }
 
-var (
-	re                 = regexp.MustCompile(`([^\\]),`)
-	apiVersionsRemover = regexp.MustCompile(`(--api-versions [^ ]+ )+`)
-)
-
 func cleanSetParameters(val string) string {
 	// `{}` equal helm list parameters format, so don't escape `,`.
 	if strings.HasPrefix(val, `{`) && strings.HasSuffix(val, `}`) {
 		return val
 	}
-	return re.ReplaceAllString(val, `$1\,`)
+
+	var result strings.Builder
+	var prevR rune
+	for i, r := range val {
+		if r == ',' {
+			if i == 0 || prevR != '\\' {
```

Comment:
I think that's a good idea, I put in a condition that i==0 in case there's a problem with the initialization of prevR, but I don't think that's necessary 😅